src/TensorNetwork.jl

```diff
@@ -45,25 +45,33 @@ function TensorNetwork{A}(tensors; meta...) where {A}
     return tn
 end
 
-function Base.replace!(tn::TensorNetwork, pair::Pair{<:Tensor, <:Tensor})
-    old_tensor, new_tensor = pair
+Base.replace(tn::TensorNetwork, old_new::Pair...) = replace!(copy(tn), old_new...)
 
-    # check if old and new tensors are compatible
-    if !issetequal(labels(new_tensor), labels(old_tensor))
-        throw(ArgumentError("New tensor labels do not match the existing tensor labels"))
+function Base.replace!(tn::TensorNetwork, old_new::Pair{<:Tensor, <:Tensor}...)
+    # Check if new tensors are already present in the network
+    new_tensors = [new_tensor for (_, new_tensor) in old_new]
+    if !isdisjoint(new_tensors, tn.tensors)
+        throw(ArgumentError("New tensors must not be already present in the network"))
     end
 
-    # update index links
-    # TODO remove this part when `Index` is removed
-    for old_label in labels(old_tensor)
-        index_obj = tn.inds[old_label]
-        link_index = findfirst(x -> x === old_tensor, index_obj.links)
-        index_obj.links[link_index] = new_tensor
-    end
+    for (old_tensor, new_tensor) in old_new
+        # check if old and new tensors are compatible
+        if !issetequal(labels(new_tensor), labels(old_tensor))
+            throw(ArgumentError("New tensor labels do not match the existing tensor labels"))
+        end
+
+        # update index links
+        # TODO remove this part when `Index` is removed
+        for old_label in labels(old_tensor)
+            index_obj = tn.inds[old_label]
+            link_index = findfirst(x -> x === old_tensor, index_obj.links)
+            index_obj.links[link_index] = new_tensor
+        end
 
-    # replace existing `Tensor` with new `Tensor`
-    index = findfirst(x -> x === old_tensor, tn.tensors)
-    tn.tensors[index] = new_tensor
+        # replace existing `Tensor` with new `Tensor`
+        index = findfirst(x -> x === old_tensor, tn.tensors)
+        tn.tensors[index] = new_tensor
```

Comment:
Use `pop!` and `push!`.

Reply:
Why? `pop!` removes the last element in the array.

Reply:
You're right. Maybe we have to revisit this idea when #55 is merged.